### DISCUSSION TITLE
cross platform

### DISCUSSION THREAD

**Discussion by carr123** [pion/webrtc#1746] [Q&A] [answered]
i find that pion is pure Go code and support many platforms .
does this mean  i have to port Go code to ARM, android, ios to implement a peer  ?
is is a problem to run Go code on android ?

**Comment by Sean-Der** [comment, marked as answer]
Pion is Pure Go, so is easy to cross compile!

If you are doing Mobile [gomobile](https://pkg.go.dev/golang.org/x/mobile/cmd/gomobile) is the best way to do it.

For a Linux server with an Intel or AMD CPU:

`CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags='-s -w'`

For a Raspberry Pi 1:

`CGO_ENABLED=0 GOOS=linux GOARCH=arm GOARM=6 go build -ldflags='-s -w'`

For a BeagleBone or a Raspberry Pi 2 or later:

`CGO_ENABLED=0 GOOS=linux GOARCH=arm GOARM=7 go build -ldflags='-s -w'`

For a 64-bit ARM board (Olimex Olinuxino-A64, Pine64, etc.):

`CGO_ENABLED=0 GOOS=linux GOARCH=arm64 go build -ldflags='-s -w'`

For a 32-bit MIPS board with no hardware floating point (WNDR3800, etc.):

`CGO_ENABLED=0 GOOS=linux GOARCH=mips GOMIPS=softfloat go build -ldflags='-s -w'`

**Comment by carr123** [comment]
thanks.
